treq/test/test_auth.py

```diff
@@ -42,6 +48,61 @@ def test_overrides_per_request_headers(self):
         )
 
 
+class PinToFirstHostAgentTests(TestCase):
+    def setUp(self):
+        self.first_agent = mock.Mock(Agent)
```

Comment:
I really dislike `mock`. Sometimes it's necessary for expedience here, but given that we already have `treq.testing`, couldn't we use that instead?